Thanks for the labels and the approval. I pushed one more commit (`38c4e32`) after your approval, which I wouldn't normally do — but I found that a sentence I'd added to this file is wrong, and the reason it's wrong also explains why both PRs are still red.

**`check conventional commits labels` cannot pass on a pull request from a fork, with or without a label.**

You labelled this PR `documentation` at 19:32:28Z. The job re-ran three seconds later and still failed. The error isn't about labels:

```
##[error]No commit found for SHA: feature/contributing-guidelines-refresh
```

That's `danielchabr/pr-labels-checker@v3.3` looking up my fork's branch name inside this repository. From its `index.cjs`:

```js
const checks = await octokit.checks.listForRef({
  ...context.repo,                            // PlaytikaOSS/pybandits
  ref: context.payload.pull_request.head.ref, // "feature/contributing-guidelines-refresh"
});
```

`context.repo` is the base repo and `head.ref` is the bare branch name, so for a same-repo branch the ref resolves and everything works. For a fork it doesn't exist here, the call throws, and the `catch` at the bottom calls `core.setFailed(error.message)` — so the job fails *after* the labels have already been evaluated successfully. The call itself is only there to tidy up stale check runs; the actual label logic above it passed.

The history matches exactly. Every internal branch passes this check (`auto_subsidy_factor` 6/6, `pybandits-neural-linear-bandit` 7/7, `limited_exploration` 3/3); my two fork branches have never passed it once, and all eight failures carry that same "No commit found for SHA" line naming the branch. #156 has been labelled `enhancement` since 16:38Z and is still red for the same reason.

v3.3 is the newest release (May 2023), so pinning forward won't help. Happy to send a small PR replacing that step with an `actions/github-script` block that reads the labels directly — it's about ten lines and has no ref lookup — but it's your CI, so only if you want it.

**What I changed here:** the paragraph I wrote said the check "re-runs automatically once the label is added", which reads as a promise that it then goes green. It re-runs, but for an outside contributor it doesn't pass. It now says the job re-runs whenever a label is added or removed, without claiming the outcome. I deliberately did **not** write the fork limitation into `CONTRIBUTING.md` — that's a bug you may well fix, and baking it into the guidelines would just make the file wrong again in the other direction.

I left the markdownlint suggestions alone. The `$` prompts and tilde fences are how every block in this file is already written, no markdownlint config or CI step exists in the repo, and changing only the blocks I touched would make the file inconsistent with itself.

Re-verified after the amend: ASCII-only, LF endings, no trailing whitespace, single trailing newline, `git diff --check` clean — checked individually, since `pre-commit` isn't installed on this machine.

--- a/.github/workflows/pull_request_style_check.yml
+++ b/.github/workflows/pull_request_style_check.yml
@@ -18,7 +18,12 @@ jobs:
     name: check conventional commits labels
     runs-on: ubuntu-latest
     steps:
-      - uses: danielchabr/pr-labels-checker@v3.3
+      - uses: mheap/github-action-required-labels@v5
         with:
-          hasSome: bug, documentation, enhancement, skip-changelog
-          githubToken: ${{ secrets.GITHUB_TOKEN }}
+          mode: minimum
+          count: 1
+          labels: |
+            bug
+            documentation
+            enhancement
+            skip-changelog

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -4,12 +4,12 @@ Contributing
 In this page we want to explain how to contribute to this library following our standards.
 
 This library aims to be an open and collaborative project released under the MIT license. It is hosted on
-[Github](https://github.com/PlaytikaResearch/pybandits) as  it intends to be supported by an open community
+[GitHub](https://github.com/PlaytikaOSS/pybandits) as  it intends to be supported by an open community
 of contributors.
 
-We want to ensure both code clarity and quality. *pybandits* follows *Flake8* standards. Every component, class and
+We want to ensure both code clarity and quality. *pybandits* follows *Ruff* standards. Every component, class and
 function are tested with *pytest* reaching a line coverage of 90%. All contributors must follow the contributors
-guidelines before to open any pull request for code merge (pre-commit hook, commits squash, *Flake8* compliance, test,
+guidelines before to open any pull request for code merge (pre-commit hook, commits squash, *Ruff* compliance, test,
 coverage, update documentation). *pybandit* also provides a detailed documentation implemented with *Sphinx* where each
 class and function is described. We want to enforce a community-based collaboration with external contributors
 that embrace our open source philosophy.
@@ -18,48 +18,50 @@ Guidelines
 ----------
 We expect developers to work with the following steps before opening a pull request.
 
-* Create a new feature branch from master. <br/> The new branch must follow the name convention
-`pybandits-XXXX feature_name` where `XXXX`= number of the issue linked to this branch and `feature_name`= short
-description of the new feature (e.g. add predict function to sMAB)
+* Create a new feature branch from `develop`, which is the default branch and the base of every pull request.
+<br/> The new branch must follow the name convention `feature/<short_description>` where `<short_description>`=
+short description of the new feature (e.g. `feature/add_predict_function_to_smab`)
 
 * Checkout on the new branch and write the code of the new feature.
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-$ git checkout feature/pybandits-XXXX feature_name
+$ git checkout develop
+$ git pull
+$ git checkout -b feature/<short_description>
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 * Run pre-commit hooks before any commit. This will check code style, max-line-length, etc. Code style must respect
-*Flake8* standards.
+*Ruff* standards (`ruff-format` and `ruff`, pinned in `.pre-commit-config.yaml`). This is the same command the CI
+style check runs, so a clean run here means a green check there.
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-$ pre-commit
+$ poetry run pre-commit run --all-files
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-* Commit and push your code on the feature branch (as many times as you need)
+* Commit and push your code on the feature branch (as many times as you need). The commit subject is an imperative
+sentence describing the change (e.g. `Add per-update decay_factor`); it carries no ticket prefix.
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 $ git add filename
-$ git commit -m 'pybandits-XXXX short_description'
+$ git commit -m 'Add short_description'
 $ git push origin HEAD
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-* Make sure that all tests must pass successfully. <br/> Tests can be executed with *pytest* running the following
-commands:
+* Make sure that all tests must pass successfully. <br/> Tests are run from the repository root with *pytest*, as the
+CI does:
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-$ cd tests
-$ pytest -vv                                      # run all tests
-$ pytest -vv test_testmodule.py                   # run all tests within a module
-$ pytest -vv test_testmodule.py -k test_testname  # run only 1 test
-$ pytest -vv -k 'not time'                        # run all tests but not exec time
+$ poetry run pytest -n auto -vv                                   # run all tests
+$ poetry run pytest -vv tests/test_testmodule.py                  # run all tests within a module
+$ poetry run pytest -vv tests/test_testmodule.py -k test_testname # run only 1 test
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 At this stage you should be at the point where you have completed all your tasks and you are ready for other people to
 review your code.
 
-* Rebase on master with commits squash. <br/> Be careful because `rebase` is an unsafe operation, if you want to know
-more please check [here](https://docs.github.com/en/get-started/using-git/about-git-rebase).
+* Rebase on `develop` with commits squash. <br/> Be careful because `rebase` is an unsafe operation, if you want to
+know more please check [here](https://docs.github.com/en/get-started/using-git/about-git-rebase).
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-$ git checkout master
+$ git checkout develop
 $ git pull
-$ git checkout feature/pybandits-<XXXX><feature_name>
-$ git rebase --interactive master
+$ git checkout feature/<short_description>
+$ git rebase --interactive develop
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 In `rebase` the first line must contain `pick`. In all the other lines delete `pick` and write `fixup`.
 Then save the file.
@@ -73,7 +75,7 @@ empty line between the title and `### Changes`; to have an empty line before the
 to have the markdown correctly rendered).
 
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-pybandits-XXXX short_description
+Add short_description
 
  ### Changes
 
@@ -85,4 +87,13 @@ pybandits-XXXX short_description
 <default message>
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-* If all steps above were successfully completed, you can open a pull request.
+* If all steps above were successfully completed, you can open a pull request against `develop`. Two conventions are
+enforced on the pull request itself:
+
+  * The title follows the same imperative form as the commit subject, and the description follows
+  [the pull request template](.github/pull_request_template.md).
+
+  * The pull request must carry at least one of the labels `bug`, `documentation`, `enhancement` or `skip-changelog`.
+  The `check conventional commits labels` job fails until one is present, and re-runs whenever a label is added or
+  removed. Contributors without write access cannot apply labels themselves, so ask a maintainer in the pull
+  request.

